phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1487,20 +1487,17 @@ private static boolean hasIndexWALCodec(Long serverVersion) {
         return MetaDataUtil.decodeHasIndexWALCodec(serverVersion);
     }
 
-    private static boolean isCompatible(Long serverVersion) {
+    private static boolean isCompatible(Long serverVersion, StringBuilder buf) {
```

Comment:
This might be out of scope of your JIRA, but it's a potentially small change. The method `isCompatible` looks pretty useless to me. Can't all the logic be moved down to `MetaDataUtil.areClientAndServerCompatible`?